app/assets/src/components/common/MetadataInput.jsx

```diff
@@ -118,7 +118,9 @@ class MetadataInput extends React.Component {
           className={className}
           onChange={val => onChange(metadataType.key, val)}
           onBlur={() => onSave && onSave(metadataType.key)}
-          value={value}
+          // If value is undefined, an empty string should be displayed.
+          // We need to explicitly pass in the empty string. Otherwise, the previous value will be retained.
+          value={isUndefined(value) ? "" : value}
```

Comment:
Why do we accept null?

Reply:
I guess the thinking is that `null` should never occur. Either the `value` prop is defined as a string, or it is undefined.

Reply:
Yeah, what Greg said. For this particular usage, we don't use `null`.

Reply:
Lol I deleted my comment because I started to have doubts. So thanks for the confirmation.

Reply:
ok, thanks. Just curious, where in the code is the non-null value guaranteed?

Reply:
Btw, if you are just worried about undefined, one other option is also to use defaultProps to set value to string.

Reply:
I've changed the code to check for null as well, and also added a more detailed comment.